libs/application-generic/src/services/resource-validator.service.ts

```diff
@@ -0,0 +1,93 @@
+import { Injectable } from '@nestjs/common';
+import { NotificationTemplateRepository } from '@novu/dal';
+import { FeatureFlagsKeysEnum } from '@novu/shared';
+
+import { GetFeatureFlag, NotificationStep } from '../usecases';
+import { ApiException } from '../utils/exceptions';
+
+@Injectable()
+export class ResourceValidatorService {
+  private readonly MAX_STEPS_PER_WORKFLOW = 10;
+  private readonly OLD_WORKFLOWS_LIMIT = 1000;
+  private readonly NEW_WORKFLOWS_LIMIT = 100;
+
+  constructor(
+    private notificationTemplateRepository: NotificationTemplateRepository,
+    private getFeatureFlag: GetFeatureFlag,
+  ) {}
+
+  async validateStepsCount(environmentId: string, steps: NotificationStep[]) {
+    const isWorkflowLimitEnabled = await this.getFeatureFlag.execute({
+      key: FeatureFlagsKeysEnum.IS_WORKFLOW_LIMIT_ENABLED,
+      environmentId,
+      organizationId: 'system',
+      userId: 'system',
+    });
+
+    if (!isWorkflowLimitEnabled) {
+      return;
+    }
+
+    if (steps.length > this.MAX_STEPS_PER_WORKFLOW) {
+      throw new ApiException({
```

Comment:
Should this be an APIException that triggers a 500? These kind of limits should trigger a 400 error so I suggest using `BadRequestException`?